gaboon/config.py

```diff
@@ -215,21 +226,24 @@ def write_dependencies(self, new_dependencies: list):
             suffix=".toml",
         )
         try:
-            temp_file.write(tomli_w.dumps(toml_data))
+            temp_file.write(tomlkit.dumps(toml_data))
             temp_file.close()
             shutil.move(temp_file.name, target_path)
         except Exception as e:
             os.unlink(temp_file.name)
             raise e
 
-        self.dependencies = new_dependencies
+        self.dependencies = dependencies
 
     def get_base_dependencies_install_path(self) -> Path:
         project_root = self._project_root
         base_install_path = project_root / self.project.get(
             DEPENDENCIES_FOLDER, DEPENDENCIES_FOLDER
         )
-        base_install_path.mkdir(exist_ok=True)
+        base_install_path.mkdir(
+            exist_ok=True,
+            parents=True,
```

Comment:
weird lint, is `--skip-magic-trailing-comma` enabled?

Reply:
https://docs.astral.sh/ruff/settings/#format_skip-magic-trailing-comma

It's defaulted to true in `ruff`. I will change. Good shout.